src/components/DashboardImmersion/ImmersionCrudForm.js

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect } from 'react'
+import PropTypes from 'prop-types'
+import * as yup from 'yup'
+import { useForm } from 'react-hook-form'
+import { yupResolver } from '@hookform/resolvers/yup'
+
+// FPCC
+import Form from 'components/Form'
+import { TYPE_WORD, TYPE_PHRASE } from 'common/constants'
+import { definitions } from 'common/utils/validationHelpers'
+
+function ImmersionCrudForm({ dataToEdit, site, submitHandler }) {
+  const validator = yup.object().shape({
+    dictionaryEntry: yup.object().required('A label is required'),
+    transKey: definitions.paragraph({ charCount: 100 }),
+  })
+
+  const defaultValues = {
+    dictionaryEntry: null,
+    transKey: '',
+  }
+
+  // NB: Not using useEditForm as we need to load the transkey value even when there is no id
```

Comment:
Maybe, I have an Idea about this. Let's chat.

Reply:
I'm marking this resolved after our chat. Filling in the blank transkey values would not be enough, since the regular edit form depends on IDs for the saved values